### ISSUE TITLE
Consumer status WARN when lag is very low

### ISSUE THREAD

**Issue by kylemurph** [linkedin/Burrow#223]
According to the consumer lag evaluation rules - https://github.com/linkedin/Burrow/wiki/Consumer-Lag-Evaluation-Rules, whenever lag increases, the status is set to WARN:

> 3. If the consumer offsets are increasing over the window, but the lag either stays the same or increases between every pair of offsets, the consumer is in a WARNING state. This means that the consumer is slow, and is falling behind.

Is there a way to set a threshold for lag in addition to this, my issue is that we trigger alerts on WARNs, but lag is very low - around 200, and clears quickly, whereas we might only want to get a WARN when lag is > 1K or so:

```
{
"error": false,
"message": "consumer group status returned",
"status": {
"cluster": "local",
"group": "CONSUMER_GROUP",
"status": "WARN",
"complete": true,
"partitions": [
{
"topic": "TOPIC",
"partition": 14,
"status": "WARN",
"start": {
"offset": 112878341,
"timestamp": 1495220705275,
"lag": 11
},
"end": {
"offset": 112882841,
"timestamp": 1495221784466,
"lag": 145
}
},
{
"topic": "TOPIC",
"partition": 23,
"status": "WARN",
"start": {
"offset": 112878287,
"timestamp": 1495220705279,
"lag": 51
},
"end": {
"offset": 112882787,
"timestamp": 1495221784464,
"lag": 185
}
}
```

**Comment by toddpalino** [comment]
No, Burrow specifically does not do lag thresholds because they are difficult to establish in a general case, and unreliable. It is designed to alert on the status of the consumer - if it is making progress, it would only warn if the lag didn't go down at some point over the interval that it collects data for.

If you want to alert on absolute lag numbers, you can use the HTTP notifier to send the data to another system that captures the total lag number and uses it with separate alerting. We do something like this internally by sending the data for all consumers, whether they are in a good or a bad state, to our metrics system (we use the threshold on the notifier to do this, by setting it to OK). You can then perform correlations such as you're suggesting.

Now, that said, it might be reasonable to add a configuration that specifies that a particular consumer should not be alerted on until it is in that state for N check cycles. In this way you could have a check interval of 10 seconds, for example, and have the count set to 6. This would mean that once a consumer gets into a bad state it would not alert for 6 * 10 seconds, or 1 minute.

**Comment by nitenA** [comment]
hi @toddpalino,

As you mentioned, we can use 
`the HTTP notifier to send the data to another system that captures the total lag number and uses it with separate alerting. We do something like this internally by sending the data for all consumers, whether they are in a good or a bad state, to our metrics system (we use the threshold on the notifier to do this, by setting it to OK`

I tried below setting with threshold to 1, but burrow only alerts for partition who has non OK status like STALL, STOP etc.. Am i missing something??

[notifier.default]
class-name="http"
url-open="http-end-point"
url-close="http-end-point"
interval=60
timeout=5
keepalive=30
extras={ api_key="ALERTS", app="PERF_burrow", tier="PERF", fabric="PERF" }
template-open="c:/src/github.com/linkedin/Burrow/config/default-http-post.tmpl"
template-close="c://src/github.com/linkedin/Burrow/config/default-http-delete.tmpl"
method-close="DELETE"
send-close=true
threshold=1

**Comment by nitenA** [comment]
Please let me know if i am missing any thing in configuration?
